python/tvm/topi/arm_cpu/mprofile/dsp/tensordot_conv2ds.py

```diff
@@ -0,0 +1,276 @@
+    x_offset = _compute_offset(data_w, kernel_w, pad_left, pad_right, stride_w)
+
+    kh_i = te.reduce_axis((0, kernel_h), name="kh_i")
+    kw_i = te.reduce_axis((0, kernel_w), name="kw_i")
+
+    padded_data = _pad_if_needed(data, "NCHW", (pad_up, pad_left, pad_down, pad_right))
+    return te.compute(
+        (batch_size, output_channels, output_h, output_w),
+        lambda n, c, y, x: te.sum(
+            padded_data[
+                n,
+                indexdiv(c, c_mul),
+                y_offset + y * stride_h + kh_i,
+                x_offset + x * stride_w + kw_i,
+            ].astype(out_dtype)
+            * kernel[indexdiv(c, c_mul), indexmod(c, c_mul), kh_i, kw_i].astype(out_dtype),
+            axis=(kh_i, kw_i),
+        ),
+        name="depthwise_conv2d",
+        tag="depthwise_conv2d_nchw_oihw_dsp",
+    )
+
+
+def _make_depthwise_conv2d_tensorization(padded_data, kernel):
+    _, _, _, padded_w = padded_data.shape
+    _, _, kernel_h, kernel_w = kernel.shape
+
+    in_dtype = padded_data.dtype
+    suffix = _get_suffix()
+    assert in_dtype == kernel.dtype
+
+    data_slice = te.placeholder((kernel_h, kernel_w), name="a", dtype=in_dtype)
+    kernel_slice = te.placeholder((kernel_h, kernel_w), name="b", dtype=in_dtype)
+
+    kh_i = te.reduce_axis((0, kernel_h), name="kh_i")
+    kw_i = te.reduce_axis((0, kernel_w), name="kw_i")
+
+    output_slice = te.compute(
+        (1,),
+        lambda k: te.sum(
+            data_slice[kh_i, kw_i].astype("int32") * kernel_slice[kh_i, kw_i].astype("int32"),
+            axis=[kh_i, kw_i],
+        ),
+        name="c",
+    )
+
+    jump = padded_w - kernel_w
+    tensordot_params = (in_dtype, kernel_h, jump, kernel_w, suffix)
+    intrin_tensordot = make_intrin_tensordot(
+        (data_slice, kernel_slice, output_slice),
+        ([padded_w, 1], [kernel_w, 1]),
+        tensordot_params,
+    )
+
+    tensordot_code = tensordot_impl(*tensordot_params)
+    return (intrin_tensordot, tensordot_code)
+
+
+def tensordot_conv2ds_schedule(_cfg, outs):
+    """Schedule function using v7e-m DSP instructions for all the conv2d operators in this file. We
+    use one schedule function for them all, because they are tensorized with the same kernel."""
+
+    schedule = te.create_schedule([x.op for x in outs])
+
+    def _callback(operator):
+        if "conv2d" in operator.tag:
+            output = operator.output(0)
+            padded_data = output.op.input_tensors[0]
+            kernel = output.op.input_tensors[1]
+
+            if operator.tag == "conv2d_nhwc_ohwi_dsp":
+                b_ax, y_ax, x_ax, co_ax = schedule[output].op.axis
+                kh_ax, kw_ax, ci_ax = schedule[output].op.reduce_axis
+                schedule[output].reorder(b_ax, y_ax, x_ax, co_ax, kh_ax, kw_ax, ci_ax)
+                intrin, code = _make_conv2d_tensorization(padded_data, kernel)
+
+            elif operator.tag == "depthwise_conv2d_nchw_oihw_dsp":
```

Comment:
are the remaining cases also tolerated?

Reply:
There should be no other remaining cases, but I've added an `else` block to raise an error if there are.